File: src/main/java/hu/indicium/dev/services/events/model/Event.java
package hu.indicium.dev.services.events.model;

import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@MappedEntity
public class Event {
    @Id
    private String id;

    private String title;

    private String description;

    private String slug;

    private LocalDateTime startDateTime;

    private LocalDateTime endDatetime;
}

File: src/main/kotlin/hu/indicium/dev/services/events/model/Event.kt
package hu.indicium.dev.services.events.model

import io.micronaut.core.annotation.Creator
import io.micronaut.core.annotation.Introspected
import org.bson.codecs.pojo.annotations.BsonCreator
import org.litote.kmongo.Id
import java.time.LocalDateTime

@Introspected
data class Event @Creator @BsonCreator constructor(
    val title: String,
    val description: String?,
    val slug: String,
    val startDateTime: LocalDateTime,
    val endDateTime: LocalDateTime,
    val _id: Id<Event>? = null
)